snuba/datasets/pipeline/query_pipeline.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABC, abstractmethod
+
+import sentry_sdk
+
+from snuba.datasets.plans.query_plan import ClickhouseQueryPlan, QueryRunner
+from snuba.request import Request
+from snuba.web import QueryResult
+
+
+class QueryPipeline(ABC):
+    """
+    Contains the instructions to build the query plans for the requested query
+    (this can be one or multiple query plans). The QueryPipeline runs the query
+    plan processors, executes the query plans and returns the relevant result.
+
+    This component is produced by the QueryPipelineBuilder.
+    """
+
+    @abstractmethod
+    def execute(self, runner: QueryRunner) -> QueryResult:
```

Comment:
I'd remove the parameter from execute so we can make the pipeline more flexible (specifically to create a pipeline for the query processing that produces another query without executing it, which is useful to recompose join queries).
You can provide the executor to the pipeline builder.

Reply:
Ok